lib/streamlit/type_util.py

```diff
@@ -233,6 +235,21 @@ def is_dataframe_like(obj: object) -> TypeGuard[DataFrameLike]:
     return any(is_type(obj, t) for t in _DATAFRAME_LIKE_TYPES)
 
 
+def is_snowpark_dataframe(obj: object) -> bool:
+    """True if obj is of type  _SNOWPARK_DF_TYPE_STR (snowflake.snowpark.dataframe.DataFrame) or
+    True when obj is a list which contains _SNOWPARK_DF_ROW_TYPE_STR (snowflake.snowpark.row.Row),
+    False otherwise"""
```

Comment:
Micro-nit: no need to say "_SNOWPARK_DF_TYPE_STR (snowflake.snowpark.dataframe.DataFrame)" - the fact that we have a string constant named `_SNOWPARK_DF_TYPE_STR` is an implementation detail that users of the API don't need to know anything about. "True if obj is a Snowpark DataFrame" would work slightly better (or "True if obj is a snowflake.snowpark.dataframe.DataFrame" works too)

Reply:
Sure 👍 It's updated now 👍